frontends/api/src/hooks/articles/queries.ts

```diff
@@ -0,0 +1,27 @@
+import { QueryOptions } from "@tanstack/react-query"
+import { articlesApi } from "../../clients"
+import type { ArticlesApiArticlesListRequest as ArticleListRequest } from "../../generated/v1"
+
+const articleKeys = {
+  root: ["articles"],
+  listRoot: () => [...articleKeys.root, "list"],
+  list: (params: ArticleListRequest) => [...articleKeys.listRoot(), params],
+  detailRoot: () => [...articleKeys.root, "detail"],
+  detail: (id: number) => [...articleKeys.detailRoot(), id],
+}
```

Comment:
The structure above (and in other queries files) is based on https://tkdodo.eu/blog/effective-react-query-keys#use-query-key-factories (blog by one of the maintainers).

Still a "key factory", but a little less automated than it was with `@lukemorales/query-key-factory`.